components/brave_wallet/browser/internal/hd_key.cc

```diff
@@ -430,6 +431,35 @@ std::vector<uint8_t> HDKey::GetPublicKeyFromX25519_XSalsa20_Poly1305() const {
   return public_key;
 }
 
+std::vector<uint8_t> HDKey::DecryptCipherFromX25519_XSalsa20_Poly1305(
+    const std::string& version,
+    const std::vector<uint8_t>& nonce,
+    const std::vector<uint8_t>& ephemeral_public_key,
+    const std::vector<uint8_t>& ciphertext) const {
+  // Only x25519-xsalsa20-poly1305 is supported by MM at the time of writing
+  if (version != "x25519-xsalsa20-poly1305")
+    return std::vector<uint8_t>();
+  if (nonce.size() != crypto_box_curve25519xsalsa20poly1305_tweet_NONCEBYTES)
+    return std::vector<uint8_t>();
+  if (ephemeral_public_key.size() !=
+      crypto_box_curve25519xsalsa20poly1305_tweet_PUBLICKEYBYTES)
+    return std::vector<uint8_t>();
+  if (private_key_.size() !=
+      crypto_box_curve25519xsalsa20poly1305_tweet_SECRETKEYBYTES)
+    return std::vector<uint8_t>();
+
+  std::vector<uint8_t> padded_ciphertext = ciphertext;
+  padded_ciphertext.insert(padded_ciphertext.begin(), 16, 0);
+  std::vector<uint8_t> padded_plaintext(ciphertext.size() + 32);
+  if (crypto_box_open(padded_plaintext.data(), padded_ciphertext.data(),
+                      padded_ciphertext.size(), nonce.data(),
+                      ephemeral_public_key.data(), private_key_.data()) != 0)
```

Comment:
Please use `const uint8_t* private_key_ptr = private_key_.data()` instead of passing raw pointer in because we can not make sure that API will not modify it. Even the interface takes const pointer now we need to defense ourselves with this so it will throw compiler error if the interface changed

Reply:
we can also pull out a getter function in case someone gets mutable raw pointer directly in the future

Reply:
I just did it via const uint8_t raw pointer. Rebased in this commit 3c4833263a173732ff44b852f4df50afb573013f

Reply:
could you also make the same change to `HDKey::GetPublicKeyFromX25519_XSalsa20_Poly1305()`?

Reply:
Done in rebased 9226d7569dd08512f14859e0a9958f15bf1928c8